nengo_loihi/simulator.py

```diff
@@ -273,8 +273,6 @@ def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_value, traceback):
-        if self.loihi is not None:
-            self.loihi.__exit__(exc_type, exc_value, traceback)
```

Comment:
We call `self.loihi.__enter__` in the `__enter__` function for this class, so I think we need to call `__exit__` too rather than just `close` (or else call neither).

If we call neither, then we're basically making the assumption that `LoihiSimulator.__enter__` does nothing, and `LoihiSimulator.__exit__` just calls `close`. I'm okay with this assumption, since I think it's generally assumed with nengo simulators of all types (for example, it's typically assumed that making a simulator and calling `close` when you're done is the same as using the context manager, though of course we encourage the context manager for its exception handling).